app/controllers/surveys.rb

```diff
@@ -12,7 +12,14 @@
   erb :'survey/show'
 end
 
-post '/surveys/new' do
-
+post '/surveys' do
+  @survey = Survey.find(params[:survey])
+    if @survey.save
+      redirect '/surveys'
+    else
+      erb :'survey/new'
```

Comment:
it would be nice to have an @errors = @survey.errors.full_messages to pass to the user so they know why they screwed up and they arent just redirected to the survey creation page again.